### PR TITLE
[Bug Fix] LPS-107347

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, @tinatian, the pr is based on https://github.com/liferay-core-infra/liferay-portal/pull/146.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by dantewang** [comment]
@yuhai Was the conflict about semantic versioning?

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by yuhai** [comment]
ci:test

**Comment by yuhai** [comment]
> @yuhai Was the conflict about semantic versioning?

yes, the conflict file was portal-impl/bnd.bnd

**Comment by tinatian** [comment]
Waiting for response of https://issues.liferay.com/browse/LPS-107347?focusedCommentId=2591808&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2591808

**Comment by tinatian** [comment]
Hi @vicnate5,
I do not understand why the expected behavior is required to be reset.

We have two ways setting up portal start from an empty database:
1.Using setup wizard like the steps here
2.Setting it manually by disabling setup wizard and modifying portal.properties

Now the behavior of 1 and 2 are different. 1 requires a password reset, 2 does not. This does not make sense to me. 

Tina

**Comment by vicnate5** [comment]
@tinatian I am asking the USM team, I am not sure why the expected result is different between Setup Wizard and no Setup Wizard

https://liferay.slack.com/archives/CNUCYF7EW/p1641437137000100

File: portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java [hunk 1]
```diff
@@ -413,9 +413,8 @@ else if (_log.isInfoEnabled()) {
 					SetupWizardSampleDataUtil.addSampleData(
 						PortalInstances.getDefaultCompanyId());
 				}
-
-				if (PropsValues.
-						SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED) {
+				else if (PropsValues.
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:416]: @dantewang why we still need this checking for SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED here? You have made SetupWizardSampleDataUtil#addSampleData read this property, isn't that enought?

Tina.

Comment (yuhai) [reply portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:416]: > @dantewang why we still need this checking for SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED here? You have made SetupWizardSampleDataUtil#addSampleData read this property, isn't that enought?
> 
> Tina.

@tinatian, @dantewang, I think the reason is that SETUP_WIZARD_ADD_SAMPLE_DATA and saving Setup Wizard are not related. When configuring Setup Wizard, we only saved SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED  (SETUP_WIZARD_ADD_SAMPLE_DATA didn't been included) to portal-setup-wizard.properties. If SETUP_WIZARD_ADD_SAMPLE_DATA is true, SetupWizardSampleDataUtil#addSampleData can read the property. If SETUP_WIZARD_ADD_SAMPLE_DATA is false, it needs to check the property SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED.

By the way, I just did push force.

Comment (dantewang) [reply portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:416]: @yuhai I have a new idea.

Maybe we can move the reset password logic to the util, and make `addSampleData` method focusing more on "sample data".

First of all, it seems `SetupWizardSampleDataUtil.addSampleData` is only called by `MainServlet`.

Secondly, if you look at setup wizard, the "Add sample data" checkbox is only checkable when database is changed; and if database is changed, we restart portal, the company and admin user should already take other properties set by setup wizard, like local, email, etc. right? If so, when adding sample data, we don't need to call `updateAdminUser` at all.

If this is agreed, we can remove a lot of parameters related to the admin user from `addSampleData`, and this commit will not be needed.

Could you check this idea?

@tinatian

**Comment by tinatian** [comment]
@yuhai @dantewang This makes sense: https://github.com/liferay-core-infra/liferay-portal/pull/158#discussion_r781156857.
So this should be a bug inside setup wizard, it should not affect the usage of it which is MainServlet. Let's try to centralize the uages of properties about setup wizard inside itself.

Please resend.

Thanks.
Tina.

**Comment by tinatian** [comment]
ci:close

**Comment by vicnate5** [comment]
Answer from USM team
https://liferay.slack.com/archives/CNUCYF7EW/p1641488270000400?thread_ts=1641437137.000100&cid=CNUCYF7EW

**Pull request closed without merging**